src/Formatter/ArrayBasedDeformatter.php

```diff
@@ -209,6 +210,8 @@ public function deserializeObject(mixed $decoded, Field $field, Deserializer $de
                 $collectingObjects[] = $propField;
             } elseif (isset($data[$propField->serializedName])) {
                 $ret[$propField->serializedName] = $deserializer->deserialize($data, $propField) ?? SerdeError::Missing;
+            } elseif (array_key_exists($propField->serializedName, $data) && $data[$propField->serializedName] === null && $propField->isNullable) {
+                $ret[$propField->serializedName] = null;
```

Comment:
These two conditional clauses could use some clarifying comments.  I just spent 5 minutes trying to figure out the difference between the new one and the one before it. :smile:   I think the first one only applies for non-null values, and the second applies for null, yes?  Do they even need to be separate conditionals at that point, since the value to assign in the latter case would be null anyway?  If the property isn't nullable, it will type error anyway later on, just like a string/int mismatch.

Reply:
This was a bit complicated, I used the `array_key_exists` without null checking, however that needed some other updated in `ArrayBasedDeformatter.php` and its interface to properly handle sequences and dictionaries.

Basically this just propagated the check there.